zilliqa/src/state.rs

```diff
@@ -382,6 +395,43 @@ impl State {
             &bincode::serialize(&account)?,
         )?)
     }
+
+    pub fn get_proof(&self, address: Address, storage_keys: &[B256]) -> Result<Proof> {
+        if !self.has_account(address)? {
+            return Ok(Proof::default());
+        };
+
+        // get_proof() requires &mut so clone state and don't mutate the origin
+        let mut state = self.clone();
+        state.root_hash()?;
+        let account = state.get_account(address)?;
+
+        let account_proof = state
+            .accounts
+            .get_proof(Self::account_key(address).as_slice())?;
+
+        let mut storage_trie = state.get_account_trie(address)?;
+        storage_trie.root_hash()?;
+
+        let storage_proofs = {
+            let mut storage_proofs = Vec::new();
+            for key in storage_keys {
+                let key = Self::account_storage_key(address, *key);
```

Comment:
@DrZoltanFazekas Are we happy that this exposes our trie encoding as part of our external API (and thus makes it hard to ever change it)?

I ask because I don't think we are encoding things into the trie the same way as Ethereum does and thus there's no guarantee that what we're doing is sensible.